packages/grafana-prometheus/src/datasource.ts

```diff
@@ -373,13 +374,30 @@ export class PrometheusDatasource
   }
 
   processTargetV2(target: PromQuery, request: DataQueryRequest<PromQuery>) {
+    // The utcOffsetSec parameter is required by the backend to ensure proper alignment of time ranges.
+    // For relative queries (e.g., Last N hours/days/weeks/years),
+    // the range must be adjusted according to the selected timezone.
+    // Without this adjustment, queries would default to using UTC-0 for "now," which may lead to incorrect results.
+    //
+    // However, this does not apply to absolute time ranges where the user explicitly sets the start and end times.
+    // In such cases, utcOffsetSec must match the user's selected timezone.
+    //
+    // - If the user relies on the browser's timezone, we derive the utcOffset from the request range object.
+    // - If the user overrides the timezone, the utcOffset must be determined based on the selected timezone.
+    //
+    // More details:
+    // - Issue that led to the introduction of utcOffsetSec: https://github.com/grafana/grafana/issues/17278
+    // - Implementation PR: https://github.com/grafana/grafana/pull/17477
+    let utcOffset = request.range.to.utcOffset();
+    if (request.timezone === 'browser') {
```

Comment:
Shouldn't this be the other way around, if it is not browser, then use the time zone? 🤔

Reply:
🤦 You are right. I went back to my notes and saw exactly what you say here in my notes. 
Thanks for that.

Reply:
https://github.com/grafana/grafana/pull/99910/commits/d47d10a90f6918460ae1aa859772140128534d88